xdp-tools: fix compilation when bpftool is installed

tk154 opened the pull request:
Currently, `xdp-tools` doesn't compile on build systems where `bpftool` is installed because additional tools and BPF programs will be compiled then, which results in build errors.

This commit disables the compilation of those `bpftool`-dependent tools.
Additionally, it drops a patch to re-add the `vlan_hdr` struct because it is required for building `xdp-filter`.

This should fix #16320.

It's currently not possible to always build the `bpftool`-dependent tools because the latest release version 7.4 of `bpftool` cannot open BPF objects that don't match the host's endianness.
But this will be possible soon, as support was merged into their main branch recently: https://github.com/libbpf/bpftool/issues/81.

Ansuel commented:
@tk154 please split this to 2 commit or maybe make another pr dropping only the patch so I can merge that faster.

tk154 committed:
xdp-tools: fix compilation when bpftool is installed

Currently, xdp-tools doesn't compile on build systems where
bpftool is installed because additional tools and BPF programs
will be compiled then, which results in build errors.

This commit disables the compilation of those bpftool-dependent tools.

Signed-off-by: Til Kaiser <mail@tk154.de>
Link: https://github.com/openwrt/openwrt/pull/16787
Signed-off-by: Robert Marko <robimarko@gmail.com>

tk154 commented:
I just split the commits and opened a new PR for dropping the patch here: #16798

robimarko commented:
Thanks! Rebased on top of main and merged!
